node/core/pvf/src/host.rs

```diff
@@ -523,7 +523,7 @@ async fn handle_execute_pvf(
 						artifact: ArtifactPathId::new(artifact_id, cache_path),
 						execution_timeout,
 						params,
-						executor_params: pvf_with_params.executor_params(),
+						executor_params: (*pvf_with_params.executor_params()).clone(),
```

Comment:
Why is `executor_params` an `Arc`? I played around with removing it and it seems to work fine, and not having an `Arc` simplified a lot of these clones etc. Or am I missing something? 🙂

https://github.com/paritytech/polkadot/compare/s0me0ne/pvf-with-executor-params...mrcnski/no-arc?expand=1

Reply:
Two considerations:
1. I think you remember why that `Arc` appeared in the first place: because we want the whole structure to be easily clonable;
2. There is already a method called `code()` (inherited from older `Pvf` struct) which returns `Arc`, and having two getters returning different types of references seems like an inconsistent design to me.

Probably you're right, and things shouldn't be overcomplicated here. I just need to either persuade myself it's okay or use some name like `executor_params_as_ref()`.

Reply:
> I think you remember why that Arc appeared in the first place: because we want the whole structure to be easily clonable

Oh right, I remember that now. It's even there in the docstring.